docs/release-instruction.md

```diff
@@ -3,70 +3,132 @@
+`README.md` if needed.
+
+### Writing changelog
+
+Add an entry to the `CHANGELOG.md` following the style established there.
+
+* copy `Unreleased` section (next steps relate to section below `Unreleased`)
+* replace `Unreleased` link with the new revision number
+* update `Unreleased...new` and `new...old` diff-links at the bottom of the file
+* add optional codename and release date in the heading
+* remove all empty sections such as `Added`, `Removed`, etc.
+* make sure all changes have references to GitHub issues in `#123` format (if possible)
+* clean up all `Unreleased` sections and leave them empty
 
-## Update VERSION files
+### Make release commit
 
-Make sure the VERSION file contains correct release version. It must be the tag
-you have just created.
+Stage changed files for commit using `git add`. Commit the changes:
 
-## Push changes and release tag to GitHub
+```shell
+$ git commit -s -m 'Release '${NEOFS_TAG_PREFIX}${NEOFS_REVISION}
+```
 
-This step should bypass the default PR mechanism to get a correct result (so
-that releasing requires admin privileges for the project), both the `master`
-branch update and tag must be pushed simultaneously like this:
+### Open pull request
 
-    $ git push origin master v0.12.0
+Push release branch:
+
+```shell
+$ git push origin release/${NEOFS_TAG_PREFIX}${NEOFS_REVISION}
+```
+
+Open pull request to the main branch of the origin repository so that the
+maintainers check the changes. Remove release branch after the merge.
+
+## Tag the release
+
+Pull the main branch with release commit created in previous step. Tag the commit
+with PGP signature.
+
+```shell
+$ git checkout master && git pull
+$ git tag -s ${NEOFS_TAG_PREFIX}${NEOFS_REVISION}
+```
+
+## Push the release tag
+
+```shell
+$ git push origin ${NEOFS_TAG_PREFIX}${NEOFS_REVISION}
+```
 
 ## Prepare and push images to a Docker Hub
 
-Create images for `neofs-storage`, `neofs-ir` and `neofs-cli`, `neofs-adm` applications
-and push them into Docker Hub (so that releasing requires privileges in nspccdev
-organization in Docker Hub)
+Create Docker images for all applications and push them into Docker Hub
+(requires [organization](https://hub.docker.com/u/nspccdev) privileges)
+
+```shell
+$ git checkout ${NEOFS_TAG_PREFIX}${NEOFS_REVISION}
+$ make images
+$ docker push nspccdev/neofs-storage:${NEOFS_REVISION}
+$ docker push nspccdev/neofs-storage-testnet:${NEOFS_REVISION}
+$ docker push nspccdev/neofs-ir:${NEOFS_REVISION}
+$ docker push nspccdev/neofs-cli:${NEOFS_REVISION}
+$ docker push nspccdev/neofs-adm:${NEOFS_REVISION}
+```
 
-    $ make images
-    $ docker push nspccdev/neofs-storage:0.24.0
-    $ docker push nspccdev/neofs-storage-testnet:0.24.0
-    $ docker push nspccdev/neofs-ir:0.24.0
-    $ docker push nspccdev/neofs-cli:0.24.0
-    $ docker push nspccdev/neofs-adm:0.24.0
+## Post-release
 
-## Make a proper GitHub release
+### Make a proper GitHub release (if not automated)
```

Comment:
mb drop at all? it is already automated

(but mb add some "check steps" since markdown in just created release could be broken)

Reply:
I left it for accident cases when automation feels bad.